src/cpp/src/visual_language/vision_encoder.hpp

```diff
@@ -86,18 +87,21 @@ class VisionEncoder {
     /// instead of the config obtained in constructors.
     /// @return Resulting embeddings for the resized source image and
     /// its slices.
-    virtual EncodedImage encode(const ov::Tensor& image, const ov::AnyMap& config_map = {}) = 0;
+    EncodedImage encode(const ov::Tensor& image, const ov::AnyMap& config_map = {});
 
     /// @brief Gets processor config
     /// @return Processor config
     ProcessorConfig get_processor_config() const;
 
 protected:
-    /// @brief A model for image encoding.
-    ov::InferRequest m_vision_encoder;
+    /// @brief  Infer requests queue for image encoding model.
+    std::unique_ptr<CircularBufferQueue<ov::InferRequest>> m_ireq_queue_vision_encoder;
+
     /// @brief A config to follow.
     ProcessorConfig m_processor_config;
 
+    virtual EncodedImage encode(ov::InferRequest& encoder, const ov::Tensor& image, const ov::AnyMap& config_map = {}) = 0;
```

Comment:
still, why do we need a overload with `ov::InferRequest` ? Can `EncodedImage encode(const ov::Tensor& image, const ov::AnyMap& config_map = {});` lock available request and execute ?

Reply:
We don't have to overload and if you think it's better to keep it the way it was before then I can change it back. 
The reason I split is that arch-specific implementation of encode requires InferRequest. Previously it had it as internal property (m_infer_request etc.). Now if we don't pass infer request to encode then we need to pull it from the queue inside every encode implementation, for every architecture we support. Even though it's like two more lines I thought it would be more straightforward to make pulling IR from the queue a common logic and developers working on support for different architectures would already have an infer request as it was before. 
Anyway it's not a big deal to change it if you prefer to have single encode method.